flink-connectors/flink-connector-kafka/src/main/java/org/apache/flink/streaming/connectors/kafka/table/KafkaDynamicSink.java

```diff
@@ -330,7 +330,7 @@ public int hashCode() {
                 topic,
                 kafkaSerializer,
                 properties,
-                FlinkKafkaProducer.Semantic.valueOf(semantic.toString()),
+                FlinkKafkaProducer.Semantic.valueOf(semantic.name()),
```

Comment:
This isn't pretty, but wasn't before either. I can of course map `KafkaSinkSemantic → Semantic` explicitly, but is there a reason these are different enums in the first place? This was already the case since I didn't create `KafkaSinkSemantic` in this PR but rather just used it in the connector options.